src/main/java/org/jenkinsci/plugins/docker/commons/credentials/DockerServerEndpoint.java

```diff
@@ -100,13 +101,16 @@ public KeyMaterialFactory newKeyMaterialFactory(@Nonnull AbstractBuild build) th
         if (workspace == null) {
             throw new IllegalStateException("Build has no workspace");
         }
-        return newKeyMaterialFactory(build.getParent(), workspace.getChannel());
+        return newKeyMaterialFactory(build, workspace.getChannel());
```

Comment:
This is wrong, my goal was to call the new undeprecated API directly, but we need an explicit cast to Run because as-written this is just an infinite loop.